### ISSUE TITLE
menu: fix axe issues

### ISSUE THREAD

**Issue by jaketrent** [pluralsight/design-system#1783]
Such as:

- Elements must only use allowed ARIA attributes (aria-allowed-attr)
 
This means that role=menuitem can't have aria-selected

And then enable the axe-core automated tests.

**Comment by jaketrent** [comment]
Missing some context here on how to proceed. Need to get history on the reason for these new components. See https://pluralsight.slack.com/archives/C5CMEKH5Z/p1623706044065300

**Comment by EdwardIrby** [comment]
We probably need some conditional logic around application of aria-selected. We could check if it's role is menu or listbox
